public/assets/ts/exam-bank.ts

```diff
@@ -110,64 +110,72 @@ class ExamBankFrontend {
+      const key = `${exam.department}-${exam.courseCode}-${exam.term}-${exam.type}`
+                  .replace("hidden", "").trimEnd();
+      let row = tableBody.querySelector(`#${key}`) as HTMLElement;
+
+      // If the row for this exam wasn't already created, then create a new row for this exam
+      // Otherwise, just update the row data with the other file for this same exam
+      if (!row) {
+        const newRow = hiddenRow.cloneNode(true) as HTMLElement;
+        newRow.id = key;
+        newRow.innerHTML = newRow.innerHTML
+          .replace(
+            "$COURSE-NAME$",
+            `${exam.department.toUpperCase()} ${exam.courseCode}`
+          )
+          .replace("$COURSE-DEPT$", exam.department)
+          .replace("$COURSE-CODE$", exam.courseCode)
+          .replace("$OFFERING$", exam.termName)
+          .replace("$TYPE$", exam.type.toLowerCase().replace("hidden", ""))
+          .replace("$EXAM-FILE$", `/exams/${exam.examFile}`)
+          .replace("$SOLUTION-FILE$", `/exams/${exam.solutionFile}`);
+
+        newRow.setAttribute("data-course-dept", exam.department);
+        newRow.setAttribute("data-course-code", exam.courseCode);
+
+        tableBody.appendChild(newRow);
+        row = newRow;
+      }
 
       const examHidden = exam.examFile?.toLowerCase().includes("-hidden");
       const solutionHidden = exam.solutionFile
         ?.toLowerCase()
         .includes("-hidden");
 
       if (exam.examFile && (this.isInAdmin || !examHidden)) {
-        newRow.querySelector(".exam-download").classList.add("active");
+        row.querySelector(".exam-download").classList.add("active");
       }
       if (exam.solutionFile && (this.isInAdmin || !solutionHidden)) {
-        newRow.querySelector(".solution-download").classList.add("active");
+        row.querySelector(".solution-download").classList.add("active");
       }
 
       // exam actions: hide and delete
       if (this.isInAdmin) {
         if (examHidden) {
-          newRow.querySelector(".exam-hide-icon").classList.add("active");
+          row.querySelector(".exam-hide-icon").classList.add("active");
         }
         if (solutionHidden) {
-          newRow.querySelector(".solution-hide-icon").classList.add("active");
+          row.querySelector(".solution-hide-icon").classList.add("active");
         }
 
-        ExamBankFrontend.populateExamActions(
-          newRow,
-          "exam",
-          exam.examFile,
-          exam
-        );
-        ExamBankFrontend.populateExamActions(
-          newRow,
-          "solution",
-          exam.solutionFile,
-          exam
-        );
-      }
+        if (exam.examFile) {
+          ExamBankFrontend.populateExamActions(
+            row,
+            "exam",
+            exam.examFile,
+            exam
+          );
+        }
 
-      if (
-        this.isInAdmin ||
-        (exam.examFile && !examHidden) ||
-        (exam.solutionFile && !solutionHidden)
-      ) {
```

Comment:
![image](https://github.com/user-attachments/assets/deb8ef07-5aed-408c-a9e7-e092abd3910c)

looks like removing these checks is causing blank rows to appear when the attached exams & solutions are hidden 😔

Reply:
Should now be fixed.